src/AppScopeTrait.php

```diff
@@ -3,5 +3,43 @@
 namespace atk4\core;
 
 trait AppScopeTrait {
+
+    public $_appScope = true;
+
+    /**
+     * Always points to current Application
+     *
+     * @var App_CLI
+     */
     public $app;
+
+    /**
+     * Object in Agile Toolkit contain $name property which is derrived from
+     * the owher object and keeps extending as you add objects deeper into
+     * run-time tree. Sometimes that may generate long names. Long names are
+     * difficult to read, they increase HTML output size but most importantly
+     * they may be restricted by security extensions such as SUHOSIN.
+     *
+     * Agile Toolkit implements a mechanism which will replace common beginning
+     * of objects with an abbreviation thus keeping object name length under
+     * control. This variable defines the maximum length of the object's $name.
+     * Be mindful that some objects will concatinate theri name with fields,
```

Comment:
It needs to be done on the object level, but the mechanics is only valid if the AppScope is present. If you think about it long names are only used for global scopes and long names there are not good.

Comment was copy-pasted from ATK, I haven't read it though ;) probably worth rewriting.
 - [ ] rewrite the comment for AppScopeTrait